### ISSUE TITLE
Strict equality does not work in all cases

### ISSUE THREAD

**Issue by oziks** [final-form/final-form-calculate#7]
### Are you submitting a **bug report** or a **feature request**?

Feature request.

### What is the current behavior?

Currently, you used a strict equality to know if you need to calculate a new value of the field. 

```js
if (next !== previous) {
  …
  form.change(…
```

### What is the expected behavior?

For some of my fields, the value is a javascript object returned by a graphQL request. The deep value of the object is the same, but the reference is not. The function is therefore executed.

Would it be possible to pass the previous values in order to reinforce the validation of the status change of the field value?

```js
if (next !== previous) {
  …
  form.change(destField, update(next, values, previousValues))
```

**Comment by oziks** [comment]
Or we could simply add a comparison attribute that would represent the comparison function. 

By default it would execute the current function : `(next, prev) => next !== prev`.

But if we want to take the comparison a step further :

```js
createDecorator(
  {
    field: 'foo',
    comparison: (next, prev) => next.id !== prev.id,
    updates: {
      doubleFoo: (fooValue, allValues) => fooValue * 2
    }
  }
```

What do you think?

**Comment by erikras** [comment]
Published fix in [`v1.1.0`](https://github.com/final-form/final-form-calculate/releases/tag/v1.1.0).
